vm/adb/adb.go

```diff
@@ -373,6 +381,32 @@ func (inst *instance) getBatteryLevel(numRetry int) (int, error) {
 	return val, nil
 }
 
+// This function was added to handled scenarios where the device being fuzzed
+// does not have a battery and is not running the battery service.
+// It first checks if the device is running a battery service.
+// If not, then the device is assumed to be battery-less (powered) and OK
+// to fuzz. The function returns without an error in that case. If the
+// device has a battery service, or a lack of service is not confirmed, then
+// the battery status is checked and the result returned.
+
+func (inst *instance) doOneBatteryCheck() ([]byte, error) {
+	out1, err1 := inst.adb("shell", "dumpsys battery | grep level:")
+	if err1 != nil {
+		// There is a possibility that the device does not have
+		// a battery. Check for that to allow such devices to be fuzzed.
+		out2, err2 := inst.adb("shell", "service check battery | grep \"not found\"")
```

Comment:
This won't work. During boot of a device with battery, `service check battery` returns `not found`.
I've run the following script during boot of a PixelC:
```
while true; do
	adb -s XXX shell 'dumpsys battery | grep level:'
	adb -s XXX shell 'service check battery'
	sleep 1
done
```
and it prints:
```
error: device '5A10000020' not found
error: device '5A10000020' not found
...
Can't find service: battery
Service battery: not found
Can't find service: battery
Service battery: not found
Can't find service: battery
Service battery: not found
...
  level: 0
Service battery: found
  level: 0
Service battery: found
  level: 0
Service battery: found
```

If you don't see another way to resolve this, then you could just set `"battery_check": false` in config.